src/components/site/Practice.jsx

```diff
@@ -19,79 +22,111 @@ class Practice extends React.PureComponent {
         ...prevState,
         data: res.data,
       }));
-      this.parseDataEcharts();
     });
   }
 
-  parseDataEcharts() {
-    const dataArr = [];
-    const { data } = this.state;
-    if (Object.keys(data).length !== 0) {
-      for (let i = 0; i < data.galaxies.length; i += 1) {
-        const point = data.galaxies[i];
-        const coor = [];
-        coor.push(point.RA);
-        coor.push(point.Dec);
-        dataArr.push(coor);
-      }
-      this.setState(prevState => ({
-        ...prevState,
-        graphData: dataArr,
-      }));
-    }
-  }
-
-  onChartClick(params){
-    console.log(params);
-    console.log(params.color)
+  onChartClick(params) {
+    this.setState(prevState => ({
+      ...prevState,
+      clickedPt: params.value,
+    }));
   }
 
   getEvent() {
     const onEvents = {
-        'click': this.onChartClick,
-    }
+      click: this.onChartClick,
+    };
     return onEvents;
-  };
+  }
 
   getOption() {
-    const { graphData } = this.state;
-    const obj = {
-      title: {
-        text: 'Exploring Observable Universe',
-      },
-      tooltip: {
-        formatter: params => {
-          const data = params.data || [0, 0];
-          return data[0].toFixed(2) + ', ' + data[1].toFixed(2);
+    const { data, clickedPt } = this.state;
+    let obj = {};
+    if (Object.keys(data).length !== 0) {
```

Comment:
Generally its poor form to create empty stuff to then mutate and return (although in some cases it will be unavoidable).  Instead of creating an empty obj, conditionally populating it, and then returning it regardless, this is a good chance to do something a lil...terser and more explicit.  One good option 
```
if (Object.keys(data).length === 0) return {};

return {
  all that options stuff
}
```
why does this work?  Similarly to one of the comments below, this leverages the fact that when javascript hits a return statement it never reaches any code past it.
why do it this way? It tells you right out front that if the stuff you want isn't there then immediately return this empty `{}`.  And says, hey, that's a special case, what I really expect to happen is this next part that's a plain ol' return not wrapped in any special conditions.  I thiiiiiiink it's a best practice to write conditional statements that catch the bad stuff and do something, rather than write more detailed conditions that describe the good stuff, as it usually results in more concise conditions.  It also saves you a nested set of `{}` which is always, always a plus.  Often when stuff gets real deeply nested it can be harder to read, and easy to accidentally drop or add brackets and stuff, I do it allll the time, and it can be a real time suck to be scrolling up and down and clicking around to figure out where you went wrong, so anytime you can take out a layer of brackets, parenthesis, etc, all the better as far as I know.